### ISSUE TITLE
Application fails without error until opening Terminal.app

### ISSUE THREAD

**Issue by jawang35** [kovidgoyal/kitty#2241]
Opening a fresh Kitty application on macOS regularly fails without any error. The behavior is as follows:
1. Open Kitty application
2. Kitty application icon bounces, indicating the application is opening
3. Kitty application icon stops bouncing. Nothing else happens.

I'm consistently able to open Kitty after opening Terminal.app first.

**Comment by Luflosi** [comment]
Open Console.app, open kitty and search for kitty in Console.app. See if there are any error messages.

**Comment by kovidgoyal** [comment]
Likely a dup of #2226

Just set the EDITOR environment variable system-wide and you will be fine.

**Comment by jawang35** [comment]
@Luflosi Console.app showed the following errors:

```
Timed out waiting for shell to quit while reading shell environment
```

```
Traceback (most recent call last):
  File "kitty/main.py", line 347, in main
  File "kitty/main.py", line 336, in _main
  File "kitty/main.py", line 270, in setup_environment
  File "kitty/main.py", line 245, in read_shell_environment
AttributeError: 'function' object has no attribute 'ans'
```

I think @kovidgoyal's hunch was correct. Setting the `editor` in `kitty.conf` fixed it. I was confused by the reference to https://github.com/kovidgoyal/kitty/pull/2226 but I did see the solution outlined in https://github.com/kovidgoyal/kitty/issues/580. Just curious why this prevents the application from opening rather than giving a runtime error when it's required or even lazily loading the `EDITOR` variable?

**Comment by Luflosi** [comment]
This bug is fixed in the latest master, see  #2232.

**Comment by crazy-matt** [comment]
Hi guys,
I turned out that this problem reappeared again on MacOS Catalina 10.15.7.  After reading the logs using the Console app, I found some "Death sentinel" traces which made me thinks about this Gatekeeper system but I'm far from being an expert of the MacOS ecosystem.
I found a way of fixing and it's persistent after a reboot. I just run from terminal the below:
`sudo xattr -dr com.apple.quarantine /Applications/kitty.app`

The only thing I'm not sure about is if you should first open the app, so if scripted running:
`open /Applications/kitty.app`
to get it quarantined and then you run the previous command to remove the quarantine flag.
I also don't know if it will be persistent after a system update.

One additional thing to mention, I'm using the editor variable of the kitty.conf to point to an absolute path to my favorite editor.
